src/aiohomeconnect/client.py

```diff
@@ -141,7 +141,9 @@ async def request(self, method: str, url: str, **kwargs: Any) -> Response:
             case codes.UNSUPPORTED_MEDIA_TYPE:
                 raise UnsupportedMediaTypeError.from_dict(response.json()["error"])
             case codes.TOO_MANY_REQUESTS:
-                raise TooManyRequestsError.from_dict(response.json()["error"])
+                err = TooManyRequestsError.from_dict(response.json()["error"])
+                err.retry_after = int(response.headers.get("Retry-After"))
```

Comment:
If we know the key is there we should use `dict[key]` instead of `dict.get(key)`.

If the key may be missing we need to handle that case to avoid a TypeError.

Reply:
Sometimes is set (time-based api rate limits) and other doesn't (amount-based rate limits)